src/cmd/manage_nsfs.js

```diff
@@ -400,23 +399,28 @@ function set_access_keys(argv, generate) {
 
 async function fetch_account_data(argv, from_file) {
     let data;
-    let generate_access_keys = true;
+    let access_keys = [{
+        access_key: undefined,
+        secret_key: undefined,
+    }];
+    let new_access_key;
     const action = argv._[1] || '';
     if (from_file) {
         const fs_context = native_fs_utils.get_process_fs_context();
         const raw_data = (await nb_native().fs.readFile(fs_context, from_file)).data;
         data = JSON.parse(raw_data.toString());
     }
     if (action !== ACTIONS.LIST && action !== ACTIONS.STATUS) _validate_access_keys(argv);
-    let new_access_key = argv.new_access_key;
-    if (action === 'update') {
-        generate_access_keys = false;
-        if (argv.regenerate) {
-            const keys = set_access_keys(argv, true);
-            new_access_key = keys[0].access_key;
-        }
+    if (action === ACTIONS.ADD || action === ACTIONS.STATUS) {
+        access_keys = set_access_keys(argv, true);
+    } else if (action === ACTIONS.UPDATE) {
+        let regenerate = false;
+        if (argv.regenerate) regenerate = true;
+        access_keys = set_access_keys(argv, regenerate);
```

Comment:
```suggestion
        access_keys = set_access_keys(argv, argv.regenerate && true);
```

Reply:
It is not equivalent.
`argv.regenerate && true` will be undefined when the `--regenerate` flag will not pass. 
you can see it by running the following snippet:
```javascript
const minimist = require('minimist');

async function main(argv = minimist(process.argv.slice(2))) {
        console.log('test argv:', argv.regenerate);
        console.log('test argv:', argv.regenerate && true);
}

main();
```

and run it by:
$node a.js --regenerate
```
test argv: true
test argv: true
```
$ node a.js
```
test argv: undefined
test argv: undefined
```

Reply:
Do you mean that false is unequal to undefined? set_access_keys() checks if (generate) so both false and undefined should work the same

Reply:
but it is "dirty", so I prefer passing false explicitly. 
more than that we explicitly define generate as boolean:
```
/**
 * set_access_keys will set the access keys either given as args or generated.
 * @param {{ access_key: any; secret_key: any; }} argv
 * @param {boolean} generate a flag for generating the access_keys automatically
 */
```

Reply:
so add boolean casting - Boolean(argv.regenerate).